File: app/src/main/java/com/motrixi/datacollection/utils/InstalledPackagesUtil.java
package com.motrixi.datacollection.utils;

import android.content.Context;
import android.content.pm.ApplicationInfo;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;

import com.adobe.fre.FREContext;

import java.util.ArrayList;
import java.util.List;

/**
 * author : Jason
 * date   : 2020/12/8 2:46 PM
 * desc   :
 */
public class InstalledPackagesUtil {

    public static ArrayList<String> getInstalledPackageList(FREContext context){
        ArrayList<String> packages = new ArrayList();
        try {
            List<PackageInfo> packageInfos = context.getActivity().getPackageManager().getInstalledPackages(PackageManager.GET_ACTIVITIES | PackageManager.GET_SERVICES);

            for (PackageInfo info : packageInfos) {


                if ((info.applicationInfo.flags & ApplicationInfo.FLAG_SYSTEM) == 0) {
                    String pkg = info.packageName;
                    String appName = (String) info.applicationInfo.loadLabel(context.getActivity().getPackageManager());
                    packages.add(pkg);
                } else {
                    //system app
                }
            }
        } catch (Throwable t) {
            t.printStackTrace();
        }
        return packages;
    }

    public static ArrayList<String> getInstalledPackageList(Context context){
        ArrayList<String> packages = new ArrayList();
        try {
            List<PackageInfo> packageInfos = context.getPackageManager().getInstalledPackages(PackageManager.GET_ACTIVITIES | PackageManager.GET_SERVICES);

            for (PackageInfo info : packageInfos) {


                if ((info.applicationInfo.flags & ApplicationInfo.FLAG_SYSTEM) == 0) {
                    String pkg = info.packageName;
                    String appName = (String) info.applicationInfo.loadLabel(context.getPackageManager());
                    packages.add(pkg);
                } else {
                    //system app
                }
            }
        } catch (Throwable t) {
            t.printStackTrace();
        }
        return packages;
    }
}

File: app/src/main/java/com/motrixi/datacollection/utils/InstalledPackagesUtil.kt
package com.motrixi.datacollection.utils

import android.content.Context
import android.content.pm.ApplicationInfo
import android.content.pm.PackageInfo
import android.content.pm.PackageManager


/**
 * author : Jason
 *  date   : 2020/10/13 5:45 PM
 *  desc   : get device IMEI
 */
object InstalledPackagesUtil {


    fun getInstalledPackageList(context: Context): List<String>? {
        val packages: MutableList<String> = ArrayList()
        try {
            val packageInfos: List<PackageInfo> = context.packageManager.getInstalledPackages(PackageManager.GET_ACTIVITIES or PackageManager.GET_SERVICES)
            for (info in packageInfos) {

                //非系统应用
                if (info.applicationInfo.flags and ApplicationInfo.FLAG_SYSTEM === 0) {
                    val pkg = info.packageName
                    val appName = info.applicationInfo.loadLabel(context.packageManager)
                    //packages.add(appName as String)
                    packages.add(info.packageName)   //包名
                }
                //系统应用
                else {

                }

            }
        } catch (t: Throwable) {
            t.printStackTrace()
        }
        return packages
    }
}